code/open_file.py

```diff
@@ -0,0 +1,42 @@
+# Actions for opening files in appropriate applications.
+from talon import Module, Context, actions, ui, app
+import os
+
+# path to knausj root directory
+REPO_DIR = os.path.dirname(os.path.dirname(__file__))
+SETTINGS_DIR = os.path.join(REPO_DIR, 'settings')
+
+mod = Module()
+ctx = Context()
+
+mod.list('talon_settings_csv', desc="Absolute paths to talon user settings csv files.")
+_csvs = {
+    x: os.path.join(SETTINGS_DIR, '_'.join(x.split()) + ".csv")
+    for x in ["file extensions", "search engines", "system paths",
```

Comment:
Nitpick - it seems unnecessarily complicated to process the names in this way, the reader has to actually inspect the code to understand what's happening.

Alternatively, the list could just contain the actual values wanted, e.g. 'file_extensions.csv'.

It would make sense to do it this way if the strings were being used as user interface elements somewhere, but that's not the case here.

Reply:
We've rewritten this to be clearer.